appinventor/blocklyeditor/src/warningHandler.js

```diff
@@ -30,6 +30,10 @@ Blockly.WarningHandler.WarningState = {
   WARNING: 1,
   ERROR: 2
 };
+Blockly.WarningHandler.prototype.warningIdHash = {};
```

Comment:
`warningIdHash` and `errorIdHash` should be initialized in the WarningHandler constructor.